[quantization] save()/load() tests and fixes

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23987 [quantization] JIT trace testing
* **#23911 [quantization] save()/load() tests and fixes**
* #23891 [quantization] jit.script() testing and fixes

Differential Revision: [D16698044](https://our.internmc.facebook.com/intern/diff/D16698044)

Commit:
[quantization] save()/load() tests and fixes

Commit:
Update on "[quantization] save()/load() tests and fixes"

Commit:
Update on "[quantization] save()/load() tests and fixes"

jamesr66a commented:
Marking as WIP since i'm adding actual invocation to the tests

Commit:
Update on "[WIP][quantization] save()/load() tests and fixes"

Commit:
Update on "[WIP][quantization] save()/load() tests and fixes"

zafartahirov commented on test/common_quantization.py:
```diff
@@ -120,6 +121,18 @@ def checkScriptable(self, mod, ref_inputs=None, ref_outputs=None):
+    def _checkScriptable(self, script_mod, ref_inputs=None, ref_outputs=None):
+        # Test save/load
+        #
+        # TODO: need __get_state__ and __set_state__
```
Instead of having a `TODO` debt, we probably should create an issue or a task

jamesr66a replied:
Created a tracking issue and added a link in the comment

Commit:
Update on "[quantization] save()/load() tests and fixes"


Differential Revision: [D16698044](https://our.internmc.facebook.com/intern/diff/D16698044)

Commit:
Update on "[quantization] save()/load() tests and fixes"


Differential Revision: [D16698044](https://our.internmc.facebook.com/intern/diff/D16698044)

Commit:
Update on "[quantization] save()/load() tests and fixes"


Differential Revision: [D16698044](https://our.internmc.facebook.com/intern/diff/D16698044)
